src/FSharpVSPowerTools.Logic/QuickInfoMargin.fs

```diff
@@ -0,0 +1,104 @@
+                title + ": " + 
+                (match errors with 
+                 | [e] -> e
+                 | _ -> 
+                    errors 
+                    |> List.mapi (fun i e -> sprintf "%d. %s" (i + 1) e) 
+                    |> List.toArray 
+                    |> String.concat " "))
+            |> String.concat " "
+
+    let flattenLines (x: string) =
+        match x with
+        | null -> ""
+        | x ->
+            x 
+            |> String.split StringSplitOptions.RemoveEmptyEntries [|"\r\n"; "\r"; "\n"|]
+            |> Array.map String.trim
+            |> Array.mapi (fun i x -> if i > 0 && x.Length > 0 && Char.IsUpper x.[0] then ". " + x else " " + x)
+            |> String.concat ""
+            |> fun x -> if x.[x.Length - 1] <> '.' then x + "." else x
+
+    let updateAtCaretPosition () =
+        let caretPos = view.Caret.Position
+        match buffer.GetSnapshotPoint caretPos with
+        | Some point ->
+            asyncMaybe {
+                let dte = serviceProvider.GetService<EnvDTE.DTE, SDTE>()
+                let! doc = dte.GetCurrentDocument(textDocument.FilePath) |> liftMaybe
+                let! project = projectFactory.CreateForDocument buffer doc |> liftMaybe
+                let! checkResults = vsLanguageService.ParseAndCheckFileInProject(textDocument.FilePath, buffer.CurrentSnapshot.GetText(), project) |> liftAsync
+                let! errors = checkResults.GetErrors() |> liftMaybe
+                do! (if Array.isEmpty errors then None else Some()) |> liftMaybe
+                return 
+                    seq { for e in errors do
+                            if String.Equals(textDocument.FilePath, e.FileName, StringComparison.InvariantCultureIgnoreCase) then
+                                match fromRange buffer.CurrentSnapshot (e.StartLineAlternate, e.StartColumn, e.EndLineAlternate, e.EndColumn) with
+                                | Some span when point.InSpan span -> yield e.Severity, flattenLines e.Message
+                                | _ -> () }
+                    |> Seq.groupBy fst
+                    |> Seq.sortBy (fun (severity, _) -> if severity = FSharpErrorSeverity.Error then 0 else 1)
+                    |> Seq.map (fun (s, es) -> s, es |> Seq.map snd |> Seq.distinct |> List.ofSeq)
+                    |> Seq.toArray
+            } 
+            |> Async.map (Option.getOrElse [||] >> updateError)
+            |> Async.StartInThreadPoolSafe
+        | _ -> updateError [||]
+
+    let docEventListener = new DocumentEventListener ([ViewChange.layoutEvent view; ViewChange.caretEvent view], 200us, updateAtCaretPosition)
+
+    interface IWpfTextViewMargin with
+        member __.VisualElement = upcast visual
+        member __.MarginSize = visual.ActualHeight + 2.
+        member __.Enabled = true
+        
+        member x.GetTextViewMargin name =
+            match name with
+            | Constants.symbolInfoMargin -> upcast x
```

Comment:
I think pattern matching always succeeds here because the literal is in lower case (see https://fslang.uservoice.com/forums/245727-f-language/suggestions/6668206-warn-when-literal-attribute-is-used-with-lowercase). Perhaps use `if/else` here instead.